### PR TITLE
Added logging of RC protocol bytes

### PR DESCRIPTION
**Pull request by tridge**
This should aid in decoding protocol varients
ping @andyp1per 

Requires ```RC_OPTIONS=16```

### CODE REVIEW

**Comment by andyp1per** [comment]
This is great, but currently doesn't help me as none of the protocol variants are able to decode the stream. I have written a "Debug" backend that simply captures the bytes and logs them in the same way if the protocol isn't detected - should enable us to see what is going on. I'll update progress on a fork of this PR.

File: libraries/AP_RCProtocol/AP_RCProtocol_Backend.cpp [hunk 1]
```diff
@@ -73,3 +74,23 @@ void AP_RCProtocol_Backend::add_input(uint8_t num_values, uint16_t *values, bool
...
 }
+
+/*
+  optionally log RC input data
+ */
+void AP_RCProtocol_Backend::log_data(AP_RCProtocol::rcprotocol_t prot, uint32_t timestamp, const uint8_t *data, uint8_t len) const
+{
+#if !APM_BUILD_TYPE(APM_BUILD_iofirmware)
+    if (rc().log_raw_data()) {
+        uint32_t u32[10] {};
```
[the remark this answers is not in the document]
Comment (andyp1per) [reply libraries/AP_RCProtocol/AP_RCProtocol_Backend.cpp]: I made this change and it works well. Interestingly it also makes it easier to spot pattern in the output. No good for decoding, but I guess python will give us that.

**Comment by andyp1per** [comment]
Here's my version:
https://github.com/ArduPilot/ardupilot/pull/13840
Need to add the script to print the output.

File: libraries/AP_RCProtocol/AP_RCProtocol_Backend.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (andyp1per) [reply libraries/AP_RCProtocol/AP_RCProtocol_Backend.cpp]: Actually I can't decode this using python - it's not valid UTF-8, maybe my python-Foo isn't good enough. I always get:
```
  File "/usr/local/lib/python2.7/dist-packages/pymavlink/DFReader.py", line 138, in __getattr__
    v = self._elements[i].decode("utf-8")
  File "/usr/lib/python2.7/encodings/utf_8.py", line 16, in decode
    return codecs.utf_8_decode(input, errors, True)
UnicodeDecodeError: 'utf8' codec can't decode byte 0xf5 in position 1: invalid start byte
```

**Comment by andyp1per** [comment]
I added a script to decode the messages, this is the data
```
0x4f, 0x8c, 0x46, 0xfa 0xa6, 0x21, 0x0e, 0x10 0x31, 0x0a, 0x01, 0x01 0x7f, 0x6b, 0x4f, 0x8c 0x7a, 0x41, 0xa6, 0x21 0x0e, 0x10, 0x40, 0x0a 0x01, 0x01, 0x7f, 0x6b 0x4f, 0x8c, 0x8d, 0x9e 0xa6, 0x21, 0x0e, 0x10 0x41, 0x0a, 0x01, 0x01
0x7f, 0x6b, 0x4f, 0x8c 0xca, 0x4d, 0xa6, 0x21 0x0e, 0x10, 0x60, 0x0a 0x01, 0x01, 0x7f, 0x6b 0x4f, 0x8c, 0xf6, 0xf6 0xa6, 0x21, 0x0e, 0x10 0x61, 0x0a, 0x01, 0x01 0x7f, 0x6b, 0x4f, 0x8c 0xb1, 0x25, 0xa6, 0x21 0x0e, 0x10, 0x70, 0x0a
0x01, 0x01, 0x7f, 0x6b 0x4f, 0x8c, 0xcb, 0x42 0xa6, 0x21, 0x0e, 0x10 0x71, 0x0a, 0x01, 0x01 0x7f, 0x6b, 0x4f, 0x8c 0x8c, 0x91, 0xa6, 0x21 0x0e, 0x10, 0x81, 0x0a 0x01, 0x01, 0x7f, 0x6b 0x4f, 0x8c, 0xc1, 0x1c 0xa6, 0x21, 0x0e, 0x10
0x10, 0x0a, 0x01, 0x01 0x7f, 0x6b, 0x4f, 0x8c 0x46, 0xfa, 0xa6, 0x21 0x0e, 0x10, 0x11, 0x0a 0x01, 0x01, 0x7f, 0x6b 0x4f, 0x8c, 0x01, 0x29 0xa6, 0x21, 0x0e, 0x10 0x21, 0x0a, 0x01, 0x01 0x7f, 0x6b, 0x4f, 0x8c 0x47, 0xf5, 0xa6, 0x21
0x0e, 0x10, 0x30, 0x0a 0x01, 0x01, 0x7f, 0x6b 0x4f, 0x8c, 0x3d, 0x92 0xa6, 0x21, 0x0e, 0x10 0x31, 0x0a, 0x01, 0x01 0x7f, 0x6b, 0x4f, 0x8c 0x7a, 0x41, 0xa6, 0x21 0x0e, 0x10, 0x40, 0x0a 0x01, 0x01, 0x7f, 0x6b 0x4f, 0x8c, 0x8d, 0x9e
0xa6, 0x21, 0x0e, 0x10 0x41, 0x0a, 0x01, 0x01 0x7f, 0x6b, 0x4f, 0x8c 0xca, 0x4d, 0xa6, 0x21 0x0e, 0x10, 0x60, 0x0a 0x01, 0x01, 0x7f, 0x6b 0x4f, 0x8c, 0xf6, 0xf6 0xa6, 0x21, 0x0e, 0x10 0x61, 0x0a, 0x01, 0x01 0x7f, 0x6b, 0x4f, 0x8c
0xb1, 0x25, 0xa6, 0x21 0x0e, 0x10, 0x70, 0x0a 0x01, 0x01, 0x7f, 0x6b 0x4f, 0x8c, 0xcb, 0x42 0xa6, 0x21, 0x0e, 0x10 0x71, 0x0a, 0x01, 0x01 0x7f, 0x6b, 0x4f, 0x8c 0x8c, 0x91, 0xa6, 0x21 0x0e, 0x10, 0x81, 0x0a 0x01, 0x01, 0x7f, 0x6b
```

File: libraries/AP_RCProtocol/AP_RCProtocol_Backend.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (tridge) [reply libraries/AP_RCProtocol/AP_RCProtocol_Backend.cpp]: yep, I've decided to leave it as u32 for now

**Comment by rmackay9** [comment]
Merged after discussion on the dev call

**Pull request merged**